### PR TITLE
feat: add new management command that will sync lms_user_id data from a temp table

### PR DESCRIPTION
**Pull request by justinhynes**
[MICROBA-1465]
- add new management command that will sync lms_user_id data from a temp table to the core_user table of the Credentials IDA

### CODE REVIEW

**Comment by justinhynes** [comment]
I didn't write tests for this management command as it is intended to run once to do the backfill. I decided it wasn't worth the time to write extensive tests. I _did_ test this locally (with a slightly modified version that loaded a local CSV over pulling data from an S3 bucket).

**Review by staubina — COMMENTED** [review]
Do you think it would be worthwhile to make it possible to run each step independently through the command line? That way we can check if each step was successful through the read replica or on the machine?

**Comment by justinhynes** [comment]
> Do you think it would be worthwhile to make it possible to run each step independently through the command line? That way we can check if each step was successful through the read replica or on the machine?

I could definitely see the benefit of breaking it up so we could mix in a bit of manual verification. Going to noodle on this a bit.

File: credentials/apps/core/management/commands/sync_lms_user_ids.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                user_id
+            FROM user_api_userretirementstatus
+        ) u
+
+    This management command assumes the data will be accessible from an AWS S3 bucket hosting the exported CSV.
+
+    This commands capabilities are split into multiple 'stages' in order to allow time for manual verifications before
+    and after syncing the LMS user data:
+        - The 'create' stage will create a temporary table and an index on the `username` column.
+        - The 'update_temp' stage will load data from a CSV containing username and
```
Comment (tuchfarber) [inline credentials/apps/core/management/commands/sync_lms_user_ids.py:50]: I think you missed some words after "and __"

File: credentials/apps/core/management/commands/sync_lms_user_ids.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            help="Name of CSV file containing the data used to populate the temp table",
+        )
+        parser.add_argument(
+            "--stage",
+            default=None,
+            help="The 'stage' of the script to run. Can be one of the following choices: 'CREATE', 'UPDATE', or "
+            "'DELETE'",
+        )
+
+    def create_temp_table_with_data(self):
```
Comment (tuchfarber) [inline credentials/apps/core/management/commands/sync_lms_user_ids.py:87]: nit: method name implies data will be loaded here. I'd probably just remove the `with_data` unless there's a reason for it.

File: credentials/apps/core/management/commands/sync_lms_user_ids.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        Loads data from a CSV file (housed in an S3 bucket) to a temporary table in the Credentials database.
+        """
+        load_data_sql = """
+            LOAD DATA FROM S3 FILE '%s/%s'
+            INTO TABLE %s
+            FIELDS TERMINATED BY ','
+            OPTIONALLY ENCLOSED BY '"'
+            LINES TERMINATED BY '\\n'
+                (username, lms_user_id)
+        """ % (
```
Comment (tuchfarber) [inline credentials/apps/core/management/commands/sync_lms_user_ids.py:129]: (params during execute)

File: credentials/apps/core/management/commands/sync_lms_user_ids.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            if max_user_id is None:
+                max_user_id = User.objects.aggregate(Max("id"))["id__max"]
+
+            batch_size = options.get("batch_size")
+            starting_user_id = options.get("starting_user_id")
+
+            self.update_credentials_user_table(batch_size, starting_user_id, max_user_id)
+        elif stage.lower() == "delete":
+            self.drop_temp_table()
+        else:
```
Comment (tuchfarber) [inline credentials/apps/core/management/commands/sync_lms_user_ids.py:209]: You can avoid this by adding a `choices` to your argument.
src: https://docs.python.org/3/library/argparse.html#choices

File: credentials/apps/core/management/commands/sync_lms_user_ids.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+                (username, lms_user_id)
+        """ % (
+            bucket_name,
+            file_name,
+            TEMP_TABLE_NAME,
+        )
+
+        with connection.cursor() as cursor:
+            logger.info("Loading temp table with LMS user data")
+            cursor.execute(load_data_sql)
```
[the remark this answers is not in the document]
Comment (justinhynes) [reply credentials/apps/core/management/commands/sync_lms_user_ids.py:137]: Good idea! I'll see if I can add this in.

**Review by tuchfarber — APPROVED** [review]

**Pull request merged**